packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -298,7 +319,14 @@ class _CupertinoPickerState extends State<CupertinoPicker> {
                 itemExtent: widget.itemExtent,
                 squeeze: widget.squeeze,
                 onSelectedItemChanged: _handleSelectedItemChanged,
-                childDelegate: widget.childDelegate,
+                childDelegate: _CupertinoPickerListWheelChildDelegateWrapper(
+                  widget.childDelegate,
+                  // If a controller is provided for this widget, it is
+                  // expected to handle its own tap to scroll behavior.
+                  onTappedChild: (widget.scrollController == null)
+                    ? (int index) => _handleChildTap(index, controller)
+                    : null,
```

Comment:
Why can't we support the user provided scroll controller?

Reply:
From a separate thread: doing so on the user provided scroll controller might conflict with their own implementation. 
I wonder if we have an example of such a conflict?

Reply:
I'm increasingly believing there is a gesture conflict that is causing tests in date picker to fail.

Take this date picker test for example, along with the changes in this PR:

<details>
<summary>With tester.drag</summary>
```dart

testWidgets('TimerPicker minDate - maxDate with minuteInterval',
      (WidgetTester tester) async {
    DateTime date = DateTime.now();
    final DateTime minimum = DateTime(2022, 6, 14, 3, 31);
    final DateTime initial = DateTime(2022, 6, 14, 3, 40);
    final DateTime maximum = DateTime(2022, 6, 14, 3, 49);
    await tester.pumpWidget(
      CupertinoApp(
        home: Center(
          child: SizedBox(
            height: 400.0,
            width: 400.0,
            child: CupertinoDatePicker(
              initialDateTime: initial,
              minimumDate: minimum,
              maximumDate: maximum,
              minuteInterval: 5,
              use24hFormat: true,
              onDateTimeChanged: (DateTime newDate) {
                date = newDate;
                print('hiii');
              },
            ),
          ),
        ),
      ),
    );

    // Drag picker minutes to min date
     await tester.drag(find.text('40'), const Offset(0.0, 1000.0), touchSlopY: 0.0);
     await tester.pumpAndSettle();

    // Returns to min date.
    expect(
      date,
      DateTime(2022, 6, 14, 3, 35),
    );
  });

```
</details>

This test fails with the changes in this PR when tester.drag is used. However, it passes when tester.trackpadFling is used instead:

<details>
<summary>With tester.trackpadFling</summary>
```dart

testWidgets('TimerPicker minDate - maxDate with minuteInterval',
      (WidgetTester tester) async {
    DateTime date = DateTime.now();
    final DateTime minimum = DateTime(2022, 6, 14, 3, 31);
    final DateTime initial = DateTime(2022, 6, 14, 3, 40);
    final DateTime maximum = DateTime(2022, 6, 14, 3, 49);
    await tester.pumpWidget(
      CupertinoApp(
        home: Center(
          child: SizedBox(
            height: 400.0,
            width: 400.0,
            child: CupertinoDatePicker(
              initialDateTime: initial,
              minimumDate: minimum,
              maximumDate: maximum,
              minuteInterval: 5,
              use24hFormat: true,
              onDateTimeChanged: (DateTime newDate) {
                date = newDate;
                print('hiii');
              },
            ),
          ),
        ),
      ),
    );

    // Drag picker minutes to min date
    await tester.trackpadFling(find.text('40'), const Offset(0.0, 1000.0), 0.1);
    await tester.pumpAndSettle();

    // Returns to min date.
    expect(
      date,
      DateTime(2022, 6, 14, 3, 35),
    );
  });

```
</details>

The documentation for `tester.trackpadFling` says:

```
A trackpad fling sends PointerPanZoom events instead of a sequence of touch events.
```

Reply:
From a separate thread:  The scrollable used by `CupertinoPicker` has default drag start behavior `DragStartBehavior.start`. The documentation for tester.drag says:

```
Breaking the drag into multiple moves is necessary for accurate execution of drag update calls with a DragStartBehavior variable set to DragStartBehavior.start. Without such a change, the dragUpdate callback from a drag recognizer will never be invoked.
```

But the tests in date_picker.dart only uses one drag move.